app/app-controllers/tabBarController.js

```diff
@@ -93,10 +93,14 @@ function(
                 this.tabBar.NavigationControllers[this.activeTabId].isActive = false;
             }
 
+            // Highlight selected tab when tab switch is
+            // made programatically
+            this.tabBar.selectItem(tabId);
             this.viewPlugin.setContentView(this.tabBar.tabViews[tabId]);
+            this.activeTabId = tabId;
```

Comment:
Was there a specific reason this line got moved up? I'm fine with re-arranging, but if there's a reason can you add a comment why it needs to be there?